### ISSUE TITLE
audioio.AudioOut – "All timers in use" on reload

### ISSUE THREAD

**Issue by raquo** [adafruit/circuitpython#9158]
### CircuitPython version

```python
Adafruit CircuitPython 9.0.2 on Adafruit ItsyBitsy M4 board
```

### Code/REPL

```python
import audioio
import board
import time

time.sleep(1)

print("INIT")

audio = audioio.AudioOut(left_channel=board.A0)

print("DONE!")
```

### Behavior

To trigger the bug, load the provided `code.py` code onto the board, and reload it 4 times (e.g. using Ctrl+D in Mu), letting it the code execute to completion each time. On the **fourth** reload, the board will fail with:

```python
code.py output:
INIT
Traceback (most recent call last):
  File "code.py", line 9, in <module>
RuntimeError: All timers in use
```

It will also fail with the same error on every second reload after this.

Disconnecting and re-connecting the board resets the "counter".

### Description

Seems similar to https://github.com/adafruit/circuitpython/issues/6061 but with AudioOut instead of PWMOut?
Note: A0 pin on ItsyBitsyt M4 has a DAC, so no PWM is involved.

**Comment by dhalbert** [comment]
> Note: A0 pin on ItsyBitsyt M4 has a DAC, so no PWM is involved.

We use a TC timer internally to time the DAC conversions. That timer is not being released properly, because there is no finalizer that runs when the `AudioOut` is gc'd. This fix would be part of #8960.

**Comment by dhalbert** [comment]
In the meantime you should be able to work around this (if it's a blocker for you), by calling `.deinit()` on the `AudioOut` when the program finishes.
